statsmodels/imputation/tests/test_mice.py

```diff
@@ -106,6 +107,26 @@ def test_default(self):
         assert_equal(exog_obs.shape, [190, 6])
         assert_equal(exog_miss.shape, [10, 6])
 
+    def test_settingwithcopywarning(self):
+        "Test that MICEData does not throw a SettingWithCopyWarning when imputing (https://github.com/statsmodels/statsmodels/issues/5430)"
+
+        df = gendat()
+        # There need to be some ints in here for the error to be thrown
+        df['intcol'] = np.arange(len(df))
+        df['intcol'] = df.intcol.astype('int32')
+
+        miceData = mice.MICEData(df)
+
+        with pd.option_context('mode.chained_assignment', 'warn'):
+            with warnings.catch_warnings(record=True) as ws:
+                warnings.simplefilter('always')
```

Comment:
would `pytest.mark.filterwarnings('ignore:pandas.core.common.is_categorical_dtype is deprecated:DeprecationWarning')` work better? you might still be `able to assert_no_warnings()`? 

You might be able to do...

```
@pytest.mark.filterwarnings('ignore:pandas.core.common.is_categorical_dtype is deprecated:DeprecationWarning')
def test_settingwithcopywarning(self):
    df = gendat()
    # There need to be some ints in here for the error to be thrown
    df['intcol'] = np.arange(len(df))
    df['intcol'] = df.intcol.astype('int32')
	
    miceData = mice.MICEData(df)
    with assert_no_warnings():
        miceData.update_all()
```

Reply:
Good thought. I just tried that though and unfortunately it doesn't work - `assert_no_warnings` still fails (must be catching them below where pytest does its filtering).